src/test/java/io/jenkins/plugins/analysis/core/quality/QualityGateEnforcerTest.java

```diff
@@ -30,27 +47,589 @@ void shouldBeSuccessfulWhenNoIssuesPresentAndNoQualityGateIsSet() {
+            softly.assertThat(fail).as(FAILURE_OVER_UNSTABLE).isEqualTo(Result.FAILURE);
+            when(run.getTotalSize()).thenReturn(0);
+
+            when(run.getTotalHighPrioritySize()).thenReturn(2);
+            Result fail2 = enforcer.evaluate(run, qualityGate);
+            softly.assertThat(fail2).as(FAILURE_OVER_UNSTABLE).isEqualTo(Result.FAILURE);
+            when(run.getTotalHighPrioritySize()).thenReturn(0);
+
+            when(run.getTotalNormalPrioritySize()).thenReturn(2);
+            Result fail3 = enforcer.evaluate(run, qualityGate);
+            softly.assertThat(fail3).as(FAILURE_OVER_UNSTABLE).isEqualTo(Result.FAILURE);
+            when(run.getTotalNormalPrioritySize()).thenReturn(0);
+
+            when(run.getTotalLowPrioritySize()).thenReturn(2);
+            Result fail4 = enforcer.evaluate(run, qualityGate);
+            softly.assertThat(fail4).as(FAILURE_OVER_UNSTABLE).isEqualTo(Result.FAILURE);
+            when(run.getTotalLowPrioritySize()).thenReturn(0);
+
+            when(run.getNewSize()).thenReturn(2);
+            Result fail5 = enforcer.evaluate(run, qualityGate);
+            softly.assertThat(fail5).as(FAILURE_OVER_UNSTABLE).isEqualTo(Result.FAILURE);
+            when(run.getNewSize()).thenReturn(0);
+
+            when(run.getNewHighPrioritySize()).thenReturn(2);
+            Result fail6 = enforcer.evaluate(run, qualityGate);
+            softly.assertThat(fail6).as(FAILURE_OVER_UNSTABLE).isEqualTo(Result.FAILURE);
+            when(run.getNewHighPrioritySize()).thenReturn(0);
+
+            when(run.getNewNormalPrioritySize()).thenReturn(2);
+            Result fail7 = enforcer.evaluate(run, qualityGate);
+            softly.assertThat(fail7).as(FAILURE_OVER_UNSTABLE).isEqualTo(Result.FAILURE);
+            when(run.getNewNormalPrioritySize()).thenReturn(0);
+
+            when(run.getNewLowPrioritySize()).thenReturn(2);
+            Result fail8 = enforcer.evaluate(run, qualityGate);
+            softly.assertThat(fail8).as(FAILURE_OVER_UNSTABLE).isEqualTo(Result.FAILURE);
+            when(run.getNewLowPrioritySize()).thenReturn(0);
+        });
+    }
+
+    /**
+     * When the gate is set to not consider 'new'-thresholds, exceeding them should
+     * still return a success.
+     */
+    @Test
+    void newWarningsShouldNotBeConsidered() {
+        QualityGateEnforcer enforcer = new QualityGateEnforcer();
+        StaticAnalysisRun run = mock(StaticAnalysisRun.class);
+        QualityGate qualityGate = new QualityGate.Builder()
+                .setNewPriorityAllUnstable(1)
+                .setNewPriorityAllFailed(1)
+                .setNewPriorityHighUnstable(1)
+                .setNewPriorityHighFailed(1)
+                .setNewPriorityNormalUnstable(1)
+                .setNewPriorityNormalFailed(1)
+                .setNewPriorityLowUnstable(1)
+                .setNewPriorityLowFailed(1)
+                .setComputeNewWarnings(false)
```

Comment:
Die Option verstehe ich nicht?

Reply:
In der Oberfläche war eine Checkbox abgebildet in der ausgewählt werden konnte, ob 'new'-Thresholds mit in die Bewertung des Builds einfließen sollen. Wurde diese vom Nutzer nicht aktiviert, so ist 'computeNewWarnings' auf false und alle 'new'-Thresholds werden ignoriert.